tests/Unit/Core/Domain/ValueObject/MoneyTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Unit\Core\Domain\ValueObject;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\Decimal\Number;
+use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\CurrencyId;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Money;
+
+class MoneyTest extends TestCase
+{
+    public function testCreateWithValidValue()
+    {
+        $money = new Money(new Number('100'), new CurrencyId(10));
+
+        $this->assertEquals('100', (string) $money->getAmount());
+    }
+
+    public function testCreateWithBadValue()
+    {
+        $this->expectException(DomainConstraintException::class);
+        $this->expectExceptionCode(DomainConstraintException::INVALID_MONEY_AMOUNT);
+
+        $money = new Money(new Number('-1'), new CurrencyId(10));
```

Comment:
O_o, why would a negative money amount be invalid?

Reply:
That's why this test checks an exception is thrown 😉

Reply:
The problem is that a negative money amount is conceptually valid (eg. a discount, a reimbursement)

Reply:
Depends on you consider it. You can consider a discount has the informations: "amount", "10", "action": "deduce". The VO Money has been introduced to handle CartRule informations, where the amount and he action are too separate things. But we can remove this validation rule very easily.